resources/measures/advanced_rtu_control/measure.rb

```diff
@@ -0,0 +1,335 @@
+					   #Add it to the unitary sys
+					   #component.resetSupplyFan()
+					   component.setSupplyFan(fan) 
+					end 
+					#handle constant speed fan objects; replace FanConstantVolume with FanVariableVolume 
+					if sup_fan.to_FanConstantVolume.is_initialized
+					   sup_fan = sup_fan.to_FanConstantVolume.get
+					   pressure_rise = sup_fan.pressureRise()
+					   #Create VS supply fan 
+					   fan = OpenStudio::Model::FanVariableVolume.new(model)
+					   fan.setName("#{air_loop_hvac.name} Fan")
+					   fan.setFanEfficiency(fan_tot_eff) 
+					   fan.setFanPowerMinimumFlowRateInputMethod("Fraction")
+					   fan.setPressureRise(pressure_rise)#keep it the same as the existing fan, since the balance of systems is the same 
+					   fan.setFanPowerMinimumFlowFraction(min_flow_fraction)
+					   fan.setMotorEfficiency(fan_mot_eff) unless fan_mot_eff.nil?
+					   #Add it to the unitary sys
+					   #component.resetSupplyFan()
+					   component.setSupplyFan(fan) 
+					end 
+				end 
+			 end 
+			 end 
+			 air_loop_hvac.thermalZones.each do |thermal_zone|
+			    min_oa_flow_rate = thermal_zone_outdoor_airflow_rate(thermal_zone) 
+				#runner.registerInfo("min_oa_flow_rate #{min_oa_flow_rate}")
+				thermal_zone.equipment.each do |equip|
+					if equip.to_AirTerminalSingleDuctConstantVolumeNoReheat.is_initialized
+				      term = equip.to_AirTerminalSingleDuctConstantVolumeNoReheat.get 
+					  runner.registerInfo("term #{term}")
+					  new_term = OpenStudio::Model::AirTerminalSingleDuctVAVHeatAndCoolNoReheat.new(model)
+						if term.autosizedMaximumAirFlowRate.is_initialized
+					     des_airflow_rate = term.autosizedMaximumAirFlowRate.get
+						 new_term.setMaximumAirFlowRate(des_airflow_rate * max_flow) #cap maximum based on overall limit 
+						 #set minimum based on max of 40% of max flow, or min ventilation level req'd 
+						 new_term.setZoneMinimumAirFlowFraction([min_flow, min_oa_flow_rate/max_flow].max)
+						elsif term.maximumAirFlowRate.is_initialized
+					      des_airflow_rate = term.maximumAirFlowRate.get
+						  new_term.setMaximumAirFlowRate(des_airflow_rate * max_flow) #cap maximum based on overall limit 
+						  #set minimum based on max of 40% of max flow, or min ventilation level req'd 
+						  new_term.setZoneMinimumAirFlowFraction([min_flow, min_oa_flow_rate/max_flow].max)
+						end 
+					  air_loop_hvac.removeBranchForZone(thermal_zone)
+					  air_loop_hvac.addBranchForZone(thermal_zone, new_term)
+					end 
+			    end
+			#handle DCV
+				if add_dcv and not ['kitchen', 'Kitchen', 'dining', 'Dining', 'Laboratory', 'KITCHEN', 'LABORATORY', 'DINING', 'patient', 'PATIENT', 'Patient'].any? { |word| (air_loop_hvac.name.get).include?(word) }
+				   oa_system = air_loop_hvac.airLoopHVACOutdoorAirSystem.get
+				   controller_oa = oa_system.getControllerOutdoorAir
+				   controller_mv = controller_oa.controllerMechanicalVentilation
+				   controller_mv.setDemandControlledVentilation(true)
```

Comment:
Should we first check to make sure the space has nonzero outdoor air before setting DCV?

Reply:
That's a good point--should I do that thru checking the design specification object?